coderd/httpapi/queryparams.go

```diff
@@ -0,0 +1,109 @@
+package httpapi
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/google/uuid"
+
+	"golang.org/x/xerrors"
+)
+
+// QueryParamParser is a helper for parsing all query params and gathering all
+// errors in 1 sweep. This means all invalid fields are returned at once,
+// rather than only returning the first error
+type QueryParamParser struct {
+	errors []Error
+}
+
+func NewQueryParamParser() *QueryParamParser {
+	return &QueryParamParser{
+		errors: []Error{},
+	}
+}
+
+// ValidationErrors is the set of errors to return via the API. If the length
+// of this set is 0, there was no errors.
+func (p QueryParamParser) ValidationErrors() []Error {
+	return p.errors
+}
+
+func (p *QueryParamParser) ParseInteger(r *http.Request, def int, queryParam string) int {
+	v, err := parse(r, strconv.Atoi, def, queryParam)
+	if err != nil {
+		p.errors = append(p.errors, Error{
+			Field:  queryParam,
+			Detail: fmt.Sprintf("Query param %q must be a valid integer (%s)", queryParam, err.Error()),
+		})
+	}
+	return v
+}
```

Comment:
Because the struct is named `QueryParamParser`, we can remove `Parse` from these prefixes IMO.